src/main/java/frc/robot/commands/TeleopSwerve.java

```diff
@@ -7,40 +29,115 @@
+        ); 
+
     }
 
     @Override
     public void execute() {
-        /* Get Values, Deadband*/
+
+        // get drive values from controller sticks
         double translationVal = MathUtil.applyDeadband(translationSup.getAsDouble(), Constants.stickDeadband);
-        double strafeVal = MathUtil.applyDeadband(strafeSup.getAsDouble(), Constants.stickDeadband);
+        double strafeVal = MathUtil.applyDeadband(strafeSup.getAsDouble(), Constants.stickDeadband) * rotationSpeed;
         double rotationVal = MathUtil.applyDeadband(rotationSup.getAsDouble(), Constants.stickDeadband);
 
-        /* Drive */
+        // set drive values to slow mode constants if slow mode is activated
+        if (slowModeSup.getAsBoolean()) {
+
+            translationVal = translationVal * Constants.SLOW_MODE_PERCENT_TRANSLATION;
+            strafeVal = strafeVal * Constants.SLOW_MODE_PERCENT_STRAFE;
+            rotationVal = rotationVal * Constants.SLOW_MODE_PERCENT_ROTATION;
+
+        }
+        
+        // rotate to score
+        if (-Constants.AUTO_ROTATE_DEADBAND <= rotationVal && rotationVal <= Constants.AUTO_ROTATE_DEADBAND && isAutoRotating) {
+            double yaw = s_Swerve.m_poseEstimator.getEstimatedPosition().getRotation().getDegrees() % 360;
+
+            // convert the yaw from [-360, 360] to [-180, 180]
+            if (yaw > 180) {
+                yaw = yaw - 360;
+            } else if (yaw < -180) {
+                yaw = yaw + 360;
+            }
+
+            // Convert to Rotation2d for rotate command
+            Rotation2d rotationYaw = Rotation2d.fromDegrees(yaw);
+
+            // Get the target rotationVal as a Rotation2d object
+            Rotation2d targetRotationVal = Rotation2d.fromDegrees(targetRotation.getAsDouble());
+
+            // Rotate rotationYaw to make the 0 of it at the target angle
+            // If targetRotationVal is 180 degrees the rotationYaw's 0 will be pointing towards 180
+            rotationYaw = rotationYaw.rotateBy(targetRotationVal);
+
+            // double error = yaw - targetRotationVal;
+            //SmartDashboard.putNumber("debug/yaw", yaw);
+            // SmartDashboard.putNumber("debug/targetRotationVal", targetRotationVal);
+            // SmartDashboard.putNumber("debug/error", error);
+            // if (Math.abs(error) > 180) {
+            //     yaw = Math.abs(360 - yaw);
+            // }
+            // SmartDashboard.putNumber("debug/calculated yaw", yaw);
+
+            SmartDashboard.putNumber("debug/calculated yaw", rotationYaw.getDegrees());
+            rotationVal = PID.calculate(rotationYaw.getDegrees(), 0.0); 
+        }
+
+        // drive swerve
         s_Swerve.drive(
             new Translation2d(translationVal, strafeVal).times(Constants.Swerve.maxSpeed), 
             rotationVal * Constants.Swerve.maxAngularVelocity, 
             !robotCentricSup.getAsBoolean(), 
-            true
+            false
```

Comment:
I talked with Everen and Alex and they both agree that they want to do open-loop driving. We have been doing that since MROC.